friends/src/components/Login/Login.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import './Login.css';
+import { connect } from 'react-redux';
+import { login } from '../../actions';
+import { Redirect } from 'react-router-dom'
+import { toast } from 'react-toastify';
+
+class Login extends Component {
+    constructor() {
+        super();
+        this.nameRef = React.createRef();
+        this.passwordRef = React.createRef();
+    }
+
+    onSubmit = (e) => {
+        e.preventDefault();
+
+        const userDetails = {
+            username: this.nameRef.current.value,
+           password: this.passwordRef.current.value
+        }
+        this.props.login( userDetails )
+    }
+
+    render() {
+        const {error } = this.props;
+
+        if(error) { 
+            toast.error(error)
+        }
+        return( 
+           <React.Fragment>
+               {this.props.isLoggedIn && <Redirect to="/friends" />}
```

Comment:
I understand what you tried to do here and it's clever, however is not working. You could have this in your App.js in this fashion `<Route path="/login" render={props => this.props.isLoggedIn ? <Redirect to"/friends" /> : <Login {...props} />} />`.
Obviously you'll need isLoggedIn slice of state passed as props.
Remember that Redirect works in synergy with Route.